app/src/main/java/org/mozilla/fenix/home/pocket/PocketCategoriesViewHolder.kt

```diff
@@ -55,12 +59,48 @@ class PocketCategoriesViewHolder(
         val categoriesSelections = components.appStore
             .observeAsComposableState { state -> state.pocketStoriesCategoriesSelections }.value
 
+        val wallpaperState = components.appStore
+            .observeAsComposableState { state -> state.wallpaperState }.value ?: WallpaperState.default
+        val wallpaperIsDefault = Wallpaper.nameIsDefault(wallpaperState.currentWallpaper.name)
+
+        val selectedBackgroundColor: Color?
+        val unselectedBackgroundColor: Color?
+        val selectedTextColor: Color?
+        val unselectedTextColor: Color?
+        when {
+            !wallpaperIsDefault &&
+                wallpaperState.currentWallpaper.cardColorLight != null &&
+                wallpaperState.currentWallpaper.cardColorDark != null -> {
+                if (isSystemInDarkTheme()) {
+                    selectedBackgroundColor = Color(wallpaperState.currentWallpaper.cardColorDark)
+                    unselectedBackgroundColor = Color(wallpaperState.currentWallpaper.cardColorLight)
+                    selectedTextColor = FirefoxTheme.colors.textActionPrimary
+                    unselectedTextColor = FirefoxTheme.colors.textActionSecondary
+                } else {
+                    selectedBackgroundColor = Color(wallpaperState.currentWallpaper.cardColorLight)
+                    unselectedBackgroundColor = Color(wallpaperState.currentWallpaper.cardColorDark)
+                    selectedTextColor = FirefoxTheme.colors.textActionSecondary
+                    unselectedTextColor = FirefoxTheme.colors.textActionPrimary
+                }
+            }
+            else -> {
+                selectedBackgroundColor = null
+                unselectedBackgroundColor = null
+                selectedTextColor = null
+                unselectedTextColor = null
+            }
+        }
+
         // See the detailed comment in PocketStoriesViewHolder for reasoning behind this change.
         if (!homeScreenReady) return
         Column {
             Spacer(Modifier.height(24.dp))
 
             PocketTopics(
+                selectedBackgroundColor = selectedBackgroundColor,
+                unselectedBackgroundColor = unselectedBackgroundColor,
+                selectedTextColor = selectedTextColor,
+                unselectedTextColor = unselectedTextColor,
```

Comment:
I brought this up in another PR review about wrapping these into some kind of object. Thoughts here? This could be a follow up naturally.

Reply:
I think this will/should get refactored as a result of:
- https://github.com/mozilla-mobile/fenix/issues/27401

Reply:
Opened:
- https://github.com/mozilla-mobile/fenix/issues/27457